src/cryptoadvance/specter/wallet.py

```diff
@@ -362,19 +362,44 @@ def fetch_transactions(self):
                 if tx.get("confirmations", 0) > 0:
                     tx["blockheight"] = current_blockheight - tx["confirmations"] + 1
         ##################### Remove until here after dropping Core v0.19 support #####################
-        self._transactions.add(txs)
         if self.use_descriptors:
-            while (
-                len(
-                    [
-                        tx
-                        for tx in self._transactions
-                        if self._transactions[tx]["category"] != "send"
-                        and not self._transactions[tx]["address"]
-                    ]
-                )
-                != 0
+            # Get all used addresses that belong to the wallet
+            addresses_info_multi = self.rpc.multi(
+                [
+                    ("getaddressinfo", address)
+                    for address in [tx for tx in txs if tx["category"] != "send"]
+                ]
+            )
```

Comment:
You don't need to do RPC calls for *all* addresses in the tx list. Some of them are already in the `wallet._addresses` and derivation index can be fetched from there instead of RPC.

Reply:
Right, fixed